js/utils/Translator.js

```diff
@@ -0,0 +1,393 @@
+(function (root, factory) {
+  if (typeof define === 'function' && define.amd) {
+    define('Translator', factory);
```

Comment:
Not sure about the name "translator" because there's not actually any translation involved. I think "Pluralizer" would be more appropriate.